Correct yard syntax for dry-initialiser classes

EdwinKruglov opened the pull request:
## Changes in this PR

- update yard documentation for dry-initializer `options` and `params`

EdwinKruglov commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Correct yard syntax for dry-initialiser classes](https://trello.com/c/2hnEtIm9/781-correct-yard-syntax-for-dry-initialiser-classes)

peterdavidhamilton reviewed (changes requested):
The indentation is inconsistent but I don't know if that matters. I'd like to see things indented under `@attribute` please, and if `default` is set, it'd be good to see that in the doc

EdwinKruglov commented:
> The indentation is inconsistent but I don't know if that matters. I'd like to see things indented under `@attribute` please, and if `default` is set, it'd be good to see that in the doc

@peterdavidhamilton I've fixed the indentation and added a note about default values where necessary. I was hoping YARD would have a `@default` tag or similar, but it doesn't, so I just mentioned the default values on the `@return` tag.
